14/umbraco-cms/customizing/extending-overview/extension-types/menu.md

```diff
@@ -4,37 +4,173 @@
 This page is a work in progress and may undergo further revisions, updates, or amendments. The information contained herein is subject to change without notice.
 {% endhint %}
 
-## Menu
-
 <figure><img src="../../../.gitbook/assets/menu.png" alt="" width="250"><figcaption><p>Menu</p></figcaption></figure>
 
-```typescript
+**JSON Manifest:**
+```json
 {
  "type": "menu",
  "alias": "My.Menu",
  "name": "My Menu"
 }
 ```
 
-### Menu Item <a href="#menu-item" id="menu-item"></a>
+**Typescript Manifest:**
+```typescript
+import { ManifestMenu } from "@umbraco-cms/backoffice/extension-registry";
+
+const menuManifest: Array<ManifestMenu> = [
+    {
+        type: 'menu',
+        alias: 'My.Menu',
+        name: 'My Menu'
+    }
+];
+```
+
+## Menu Item
 
 <figure><img src="../../../.gitbook/assets/menu-item.png" alt="" width="250"><figcaption><p>Menu Item</p></figcaption></figure>
 
-```typescript
+Menu items are the items that appear in the menu. 
+
+To add custom menu items, you can define a single MenuItem manifest and link an element to it. In this element, you can fetch the data and render as many menu items as you want based on that data.
+
+The code snippets below show how to declare a new menu item using JSON or Typescript.
+
+### JSON Manifest
+
+```json
 {
  "type": "menuItem",
  "alias": "My.MenuItem",
  "name": "My Menu Item",
+ "element": "./menu-items.ts",
  "meta": {
   "label": "My Menu Item",
   "menus": ["My.Menu"]
  }
 }
 ```
 
-### **Tree Menu Item**
+### Typescript Manifest
+
+```typescript
+const menuItemManifest: Array<ManifestMenuItem> = [
+    {
+        type: 'menuItem',
+        alias: 'My.MenuItem',
+        name: 'My Menu Item',
+        meta: {
+            label: 'My Menu Item',
+            menus: ["My.Menu"]
+        },
+        element: () => import('./menu-items.ts')
+    }
+];
+```
+
+### The UI Element
```

Comment:
I've changed this from "The Lit Element" to "The UI Element".

Makes it a little more generic, and easier to understand for those not familiar with Lit.
What do you think?

Reply:
I will look in to it tonight!